src/peergos/shared/util/AsyncLock.java

```diff
@@ -38,6 +38,7 @@ public AsyncLock(CompletableFuture<T> initialValue) {
                     updater.get().thenCompose(processor)
                             .thenApply(res -> newHead.complete(res) && result.complete(res))
                             .exceptionally(e -> newHead.complete(current) && result.completeExceptionally(e));
+                    t.printStackTrace();
```

Comment:
are you happy to leave stacktrace call in here?

Reply:
Yep this took me ages to find because there was otherwise no log of the error